### PR TITLE
Moved the AssetBrowserFileEndingsFilter to IGlobalModData

### LINKED ISSUE #17523: Move AssetBrowserLogic SupportedFormats
Every mod has to duplicate the `chrome/assetbrowser.yaml` for the line `SupportedFormats: ` which is a bit of a non-sense. I suggest to move it to `metrics.yaml` which seems to be a general dumping ground as it also contains `ColorPickerActorType` used for the asset browser.

### PR DESCRIPTION
**Pull request by Mailaender**
Closes https://github.com/OpenRA/OpenRA/issues/17523.

### CODE REVIEW

File: OpenRA.Mods.Common/Widgets/Logic/AssetBrowserLogic.cs [hunk 1]
```diff
@@ -214,8 +214,12 @@ public AssetBrowserLogic(Widget widget, Action onExit, ModData modData, World wo
 				prevButton.IsVisible = () => !isVideoLoaded;
 			}
 
-			if (logicArgs.ContainsKey("SupportedFormats"))
-				allowedExtensions = FieldLoader.GetValue<string[]>("SupportedFormats", logicArgs["SupportedFormats"].Value);
+			var hasssetBrowserFileEndingsFilter = modData.Manifest.Contains<AssetBrowserFileEndingsFilter>();
```
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/Logic/AssetBrowserLogic.cs]: This check isn't needed - `Manifest.Get` will create an instance of the module with default values if it hasn't been explicitly defined in `mod.yaml`.

If you change the default value for `AssetBrowserFileEndingsFilter.SupportedFormats` to an empty array then you should be able to do just
```csharp
allowedExtensions = modData.Manifest.Get<AssetBrowserFileEndingsFilter>().SupportedFormats;
```
here.

File: OpenRA.Mods.Common/AssetBrowserFileEndingsFilter.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+namespace OpenRA
+{
+	public class AssetBrowserFileEndingsFilter : IGlobalModData
```
Comment (pchote) [inline OpenRA.Mods.Common/AssetBrowserFileEndingsFilter.cs]: Just `AssetBrowser` would be better. We don't gain anything by making the name more specific, and it prevents us from adding other asset browser related things here in the future.

File: OpenRA.Mods.Common/AssetBrowserFileEndingsFilter.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+	{
+		[FieldLoader.Require]
+		public readonly string[] SupportedFormats = null;
```
Comment (pchote) [inline OpenRA.Mods.Common/AssetBrowserFileEndingsFilter.cs]: Then this should really be `SupportedExtensions` to reflect how it is used.

**Comment by Mailaender** [comment]
Updated.

**Comment by Mailaender** [comment]
Pinging @IceReaper who may also like this due to
https://github.com/IceReaper/KKnD/blob/5fc5d83ac2ccdb640a1d773e7eaa998f2fff5069/mods/kknd/chrome/assetbrowser.yaml#L1-L3

**Review by pchote — APPROVED** [review]

**Review by abcdefg30 — APPROVED** [review]

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/ca350b03d75eec30aea2e99739c11c198c1249e9...48208a712ac54b24cc44f3373fc060b124bee3dc)

**Comment by abcdefg30** [comment]
[SDK update notes](https://github.com/OpenRA/OpenRAModSDK/wiki/Update-notes:-hotfix-release-to-next-release/_compare/bdcda92ed1234726505ea5c9cb9e30d61b1bdbfd...3ce2c2661125d8caf71fe52f01033a108575f79c)

**Pull request merged**